File: clients/python/diom/models/auth_token_expire_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class AuthTokenExpireIn(BaseModel):
    namespace: t.Optional[str] = None

    id: str

    expiry_millis: t.Optional[int] = Field(default=None, alias="expiry_millis")
    """Milliseconds from now until the token expires. `None` means expire immediately."""

File: z-clients/rust/src/models/auth_token_expire_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AuthTokenExpireIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub id: String,

    /// Milliseconds from now until the token expires. `None` means expire immediately.
    #[serde(
        rename = "expiry_ms",
        skip_serializing_if = "Option::is_none",
        with = "crate::duration_ms_serde::optional"
    )]
    pub expiry: Option<std::time::Duration>,
}

impl AuthTokenExpireIn {
    pub fn new(id: String) -> Self {
        Self {
            namespace: None,
            id,
            expiry: None,
        }
    }

    pub fn with_namespace(mut self, value: impl Into<Option<String>>) -> Self {
        self.namespace = value.into();
        self
    }

    pub fn with_expiry(mut self, value: impl Into<Option<std::time::Duration>>) -> Self {
        self.expiry = value.into();
        self
    }
}
